File: 11-May Flood Fill/Flood_Fill_1.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        rows = len(image)
        cols = len(image[0])
        init  = image[sr][sc]
        
        def helper(i, j, newColor, init):
            if (i<0 or i>=rows or j<0 or j>=cols or image[i][j]!=init or image[i][j]==newColor):
                return 
            image[i][j] =  newColor

            helper(i+1, j, newColor, init)
            helper(i-1, j, newColor, init)
            helper(i, j+1, newColor, init)
            helper(i, j-1, newColor, init)

        helper(sr,sc,newColor, init)
        return image

File: 11-May Flood Fill/Flood_Fill_1.cpp
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        int k = image[sr][sc];
        if(k == newColor)
            return image;
        flood(image, sr, sc, newColor, k);
        return image;
    }
        

    void flood(vector<vector<int>>& image,int sr,int sc,int newColor,int k){
        if (0 <= sr and sr < image.size() and 0 <= sc and sc < image[0].size()){
            if(image[sr][sc] == k){
                image[sr][sc] = newColor;
                flood(image, sr+1, sc, newColor, k);
                flood(image, sr, sc+1, newColor, k);
                flood(image, sr-1, sc, newColor, k);
                flood(image, sr, sc-1, newColor, k);
            }
        }
    }
};
